### DISCUSSION TITLE
Power Spectral Density

### DISCUSSION THREAD

**Discussion by gnssLa** [groops-devs/groops#90] [Q&A] [answered]
Hi GROOPS Developers,

I am interested in Power Spectral Density (PSD) using GROOPS. However, I am not sure about how I can use the **[Instrument2PowerSpectralDensity](https://groops-devs.github.io/groops/html/Instrument2PowerSpectralDensity.html)**, **[InstrumentSetType](https://groops-devs.github.io/groops/html/InstrumentSetType.html)**, and **[NoiseInstrument](https://groops-devs.github.io/groops/html/NoiseInstrument.html)**. in order to come up with the PSD plot using the **[PlotGraph.](https://groops-devs.github.io/groops/html/PlotGraph.html)**

Could you provide me with a sample script for this?

Thanks for your support.

Regards,

Bax.

**Comment by zhedumi** [comment, marked as answer]
Hi,

I recomment checking out [InstrumentDataHandling cookbook](https://groops-devs.github.io/groops/html/cookbook.instrument.html) which show ssome examples on how to work with data, plotting and PSD generation within groops.

Simply said you process the instrument data you want to analyze with [Instrument2PowerSpectralDensity](https://groops-devs.github.io/groops/html/Instrument2PowerSpectralDensity.html) and plot the output file with [PlotGraph.](https://groops-devs.github.io/groops/html/PlotGraph.html). 

Also we recently added [tutorial videos](https://www.youtube.com/channel/UCzUirVN2JuEKZwdM9dIqllg/videos) for groops which also shows how to handle data in groops.

Best regards,

Patrick

**Comment by gnssLa** [comment]
Hi,

I checked the [InstrumentDataHandling cookbook](https://groops-devs.github.io/groops/html/cookbook.instrument.html) before asking the question.  I also watched the [tutorial videos](https://www.youtube.com/channel/UCzUirVN2JuEKZwdM9dIqllg/videos). I have learned and I am still learning from the documentation.

I will check the documentation, lest I missed something regarding this.
![example](https://user-images.githubusercontent.com/109293325/183329709-e39c5696-735e-4c16-a68d-da074776a8fc.PNG)
 
Thanks.

**Comment by Waley04** [comment]
@gnssLa,

Are you looking for something like this? 

![graz_2020-01-01_spectral](https://user-images.githubusercontent.com/101952777/183342619-bb5414c6-0aef-4a00-93fc-7af25ee98ffd.png)

It depends on the instrument data type. 

You also refer to:

=> [Instrument2CovarianceFunctionVCE](https://groops-devs.github.io/groops/html/Instrument2CovarianceFunctionVCE.html), =>[CovarianceFunction2PowerSpectralDensity](https://groops-devs.github.io/groops/html/CovarianceFunction2PowerSpectralDensity.html), =>[PowerSpectralDensity2CovarianceFunction](https://groops-devs.github.io/groops/html/PowerSpectralDensity2CovarianceFunction.html).

Cheers.

**Comment by zhedumi** [comment]
Hi,

@gnssLa then I may missunderstood the question. What exactly is the Problem? Do you have a timeseries to be analyzed and you are not sure how to accomplish this? Is it just about the formating in the graph. 
As @Waley04 the processing of the data varies a bit depending on the instrument data. 

Best regards.

**Comment by gnssLa** [comment]
@zhedumi,

I have the time series data to be analysed. However, I am not sure about how to it in GROOPS. 

Here is an example with L1P, L2I, L5P, and L6I observations.  
**dgar_L1PL2IL5PL6I :** [dgar_L1PL2IL5PL6I.txt](https://github.com/groops-devs/groops/files/11900577/dgar_L1PL2IL5PL6I.txt)

Questions: 
1. How can I plot the Allan Deviation vs Averaging Interval? 
2. If the given file is not proper, could you give me an example?

**General Purpose includes**

Determination of
1. frequency stability, 
2. clock prediction accuracy, and 
3. clock rate variation stability

Thanks in advance.

Regards.

**Reply by zhedumi** [reply]
Hi,

For the allan deviation you might be interested into the [Instrument2AllanVariance](https://groops-devs.github.io/groops/html/Instrument2AllanVariance.html) than and you'd just need to plot the 0 column of the result to the other columns or so I would assume atleast.

For the determination of the clock qualities I am not very experienced in what mathematical analysis tools are used so I'd probably would need to know what the usual processing chain is before I know what groops programs to recommend.

Best regards,

Patrick

**Comment by gnssLa** [comment]
@zhedumi ,

Thanks for your response. I already processed the [Instrument2AllanVariance](https://groops-devs.github.io/groops/html/Instrument2AllanVariance.html) and here is the result:

[example_allan.txt](https://github.com/groops-devs/groops/files/11906711/example_allan.txt)

I am not sure about your direction on plotting. How can I get a plot of the Allan Deviation for all satellites? 

Thanks.
